src/traffic_server/CMakeLists.txt

```diff
@@ -80,6 +80,9 @@ endif (TS_USE_LINUX_IO_URING)
 if (TS_USE_LINUX_NATIVE_AIO)
     target_link_libraries(traffic_server PRIVATE aio)
 endif (TS_USE_LINUX_NATIVE_AIO)
+if(TS_USE_POSIX_CAP)
+    target_link_libraries(traffic_server PUBLIC cap::cap)
+endif()
```

Comment:
I don't think traffic_server has a direct dependency on libpcap, so it should not be linked here. Let CMake take care of resolving the transitive dependencies.

```cmake
add_library(dependency dependency.cc)

target_link_libraries(dependency PUBLIC transitive::transitive)

add_executable(main main.cc)

target_link_libraries(main PRIVATE dependency)
```

CMake will link main to transitive::transitive if dependency is STATIC; if dependency is SHARED, then include paths for transitive::transitive will be added to main, but the library will not be linked.

This is the ideal. We can't do it everywhere in ATS yet because our dependency graph is a nightmare; if we really have to link directly to the transitive dependency for now, let's mark it so we know it should be fixed later.

This applies to a number of targets changed in this PR.

Reply:
I fixed this one.  I'm sure there are more link commands that are over specified, but we can address those in a future pass as we further consolidate targets.